### PR TITLE
Cherry pick #465 in cloud provider azure: Cleanup subnet in frontend IP configs

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
Cleanup subnet in frontend IP configs to prevent huge subnet request bodies in some scenarios.
Ref: https://github.com/kubernetes-sigs/cloud-provider-azure/pull/465

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Cleanup subnet in frontend IP configs to prevent huge subnet request bodies in some scenarios.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/priority critical-urgent
/area provider/azure
/sig cloud-provider
/triage accepted

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
/assign @cheftako  @andrewsykim 
for cloud provider approve

**Comment by liggitt** [comment]
redirecting, I'm only in the approvers list for dep updates

/unassign
/assign andrewsykim cheftako

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go [hunk 1]
```diff
@@ -183,11 +183,40 @@ func (az *Cloud) CreateOrUpdateSecurityGroup(sg network.SecurityGroup) error {
 	return rerr.Error()
 }
 
+func cleanupSubnetInFrontendIPConfigurations(lb *network.LoadBalancer) network.LoadBalancer {
+	if lb.LoadBalancerPropertiesFormat == nil || lb.FrontendIPConfigurations == nil {
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:187]: What's the reason for checking if `LoadBalancerPropertiesFormat` is nil? I don't see it used below

Comment (cheftako) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:187]: +1

Comment (nilo19) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:187]: `lb.FrontendIPConfigurations` is actually `lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations `

**Comment by feiskyer** [comment]
ping @andrewsykim. Could you help to approve this change?

**Pull request merged**